### PR TITLE
Finish autoreset GUI integration

### PR DESCRIPTION
**Pull request by jakob223**
Issue: Y2K-714
cc @keybase/y2ksquad 

This is not quite ready, it's got a few bugs still. I'm having trouble testing it effectively. Putting up this PR so that other people can also be working on it, maybe.

Misc existing bugs:

- [ ] the login.login saga is not getting closed
- [x] back button not working on confirm screen
- [ ] when cancelling the reset is in progress, the user should get some feeedback, e.g. a spinner
- [x] TSC is not passing because I did something wrong w/r/t the saga situation in actions/recover-password
 - [x] can't close Waiting screen
 - [x] different things pop up before it settles on a time on waiting screen

### CODE REVIEW

**Comment by jakob223** [comment]
I would like to get a review on this even though it isn't finished because it's almost finished and ideally it can get merged very soon so that QA can happen on autoresets.

**Comment by jakob223** [comment]
ok, @buoyad I think I'm ready for another review on this. It's got one or two things to iron out (the "a bit" that appears for a second, and the login saga not closing), but I think it's pretty much ready to go.

File: shared/constants/autoreset.tsx [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import {formatDurationForAutoreset as formatDuration} from '../util/timestamp'
 
-export const autoresetEnterPipelineWaitingKey = 'autoreset:EnterPipelineWaitingKey'
+export const waitingKeyEnterPipeline = 'autoreset:EnterPipelineWaitingKey'
+export const waitingKeyActuallyReset = 'autoreset:ActuallyResetWaitingKey'
 export const waitingKeyCancelReset = 'autoreset:cancelWaitingKey'
```
Comment (buoyad) [inline shared/constants/autoreset.tsx]: make `waitingKey` a suffix please

File: shared/constants/types/autoreset.tsx [hunk 2]
```diff
@@ -2,6 +2,7 @@ export type State = Readonly<{
   active: boolean
   endTime: number
   error: string
+  hasWallet: boolean
```
Comment (buoyad) [inline shared/constants/types/autoreset.tsx]: 🎉 works

File: shared/login/reset/waiting.tsx [hunk 3]
```diff
@@ -24,7 +22,23 @@ const Waiting = (props: Props) => {
...
+  React.useEffect(() => {
+    function tick() {
+      const newFormattedTime = Constants.formatTimeLeft(endTime)
+      if (formattedTime !== newFormattedTime) {
+        setFormattedTime(newFormattedTime)
+      }
+    }
+
+    const tickerID = addTicker(tick)
+    return function cleanup() {
```
Comment (buoyad) [inline shared/login/reset/waiting.tsx]: no need to name, `return () => removeTicker(tickerID)`

File: shared/login/reset/waiting.tsx [hunk 4]
```diff
@@ -24,7 +22,23 @@ const Waiting = (props: Props) => {
...
+      removeTicker(tickerID)
+    }
+  })
```
Comment (buoyad) [inline shared/login/reset/waiting.tsx]: I think this needs a `deps` array or will render thrash.

File: shared/login/reset/waiting.tsx (hunk 3 above)
Comment (jakob223) [reply shared/login/reset/waiting.tsx]: this pattern is just so illegible to me without the name :/

**Comment by zapu** [comment]
:tada:

**Pull request merged**